TTS code refactoring

delee-dix opened the pull request:
• 비즈니스 로직과 화면 UI를 나누는 작업
→ 정상 동작 X
→ voice 쪽에 error 발생

delee-dix committed:
code Refactoring (tts)

tts sample code refactoring → 정상 작동 확인 X

bwlee-dix commented on lib/screen/tts_example.dart:
```diff
@@ -205,161 +33,507 @@ class _TtsSampleAppState extends State<TtsSampleApp> {
-    );
-  }
 }
+
```
주석 부분은 백업을 위해 필요하다면 추후 git으로 추적할 수 있으니,
리뷰가 필요한 상황에서는 제거해주셔도 좋을 것 같아요!

delee-dix replied:
넵! 앞으로도 그렇게 하도록 하겠습니다! 주석 부분은 삭제 처리 완료 하였습니다!

bwlee-dix commented on lib/service/tts_example.dart:
```diff
@@ -0,0 +1,89 @@
+    if (Platform.isAndroid) {
+      _getDefaultEngine();
+      _getDefaultVoice();
+    }
```
_getDefualtEngine, _getDefaultVoice은 안드로이드에서만 사용 가능한 함수일까요?!

delee-dix replied:
네 안드로이드에서만 사용이 가능한 함수입니다! 'flutter_tts.dart' 파일 내에도 ***Android supported only*** 라는 주석이 존재했습니다!

bwlee-dix commented on lib/service/tts_example.dart:
```diff
@@ -0,0 +1,89 @@
+    print("Default voice data: $language");
+    language = await flutterTts.getDefaultVoice;
+  }
+
```
함수명에 언더바(_)의 유무는 어떤 기준으로 나뉘었을까요?!

e.g., `_getDefaultVoice()` vs `speak()`

delee-dix replied:
원래 언더바(_)는 private한 함수나 변수가 필요할 때 사용하는 것으로 알고 있습니다!
이번 경우에 있어서도 내부에서만 사용하는 함수와 외부에 참조하는 함수를 나누기 위해서 사용하였습니다!

bwlee-dix commented on lib/screen/tts_example.dart:
```diff
@@ -205,161 +33,507 @@ class _TtsSampleAppState extends State<TtsSampleApp> {
-              _buildSliders(),
-              if (isAndroid) _getMaxSpeechInputLengthSection(),
+            children: <Widget>[
+              // _inputSection(),
```
리팩토링 과정에서 남겨진 쓰이지 않을 함수에 대한 주석 같아보이는데, 
제거 하셔도 좋을 것 같습니다! 

- _inputSection()
- _btnSection()
- _engineSection() 등

delee-dix replied:
앗! 해당 영역에 주석을 사용하여 구분하기 쉽게 하려던 것이었는데 이름을 함수명처럼 작성을 했네요!!
_inputSection() = inputSection 이라고 작성을 하려는 의도였습니다!
해당 작업도 정리하였습니다!

bwlee-dix commented on lib/screen/tts_example.dart:
```diff
@@ -205,161 +33,507 @@ class _TtsSampleAppState extends State<TtsSampleApp> {
+                  mainAxisAlignment: MainAxisAlignment.spaceEvenly,
+                  children: <Widget>[
+                    ElevatedButton(
+                      onPressed: () => ttsService.speak,
```
speak에 '소괄호' (`()`, `함수 호출 연산자`)가 누락되어 호출이 안됐을 것 같습니다! 
하단의 stop, pause도 동일한 문제가 있을 거라 예상됩니다!


케이스 | 권장 여부 | 설명
-- | -- | --
ttsService.speak | ✅ | 함수 참조를 직접 전달, 추가 로직이 필요하지 않을 때 사용.
() => ttsService.speak() | ✅ | 호출 시 추가 로직이나 데이터를 전달해야 할 경우 사용
() => ttsService.speak | ❌ | 호출되지 않고 참조만 반환하므로 기능 수행 X

delee-dix replied:
아하! 작성해주신 케이스를 확인하니 이해가 쉽게 되었습니다! 감사합니다!!

bwlee-dix commented:
View와 Service를 확실히 분리해주셔서, 비즈니스 로직만 살펴보기 훨씬 좋네요! 

PR 코멘트에 정상 작동되지 않는다고 작성해주셨는데, 
View에 연결한 이벤트 핸들러에 ()를 붙여서 한번 테스트 해주시고, 경과 알려주세요! 
또 다른 문제가 있다면 커밋 푸시 후 코멘트 남겨주세요!

고생 많으셨어요!! 😄

bwlee-dix commented on lib/screen/tts_example.dart:
```diff
@@ -205,161 +33,507 @@ class _TtsSampleAppState extends State<TtsSampleApp> {
+                            ),
+                          )
+                          .toList(),
+                      onChanged: ttsService.changedEngines,
```
`onChanged`으로 콜백 함수를 실행시킬 때, 
UI 상태를 업데이트 할 수 있도록 로직을 추가하여 해결 할 수 있었습니다!

``` dart
onChanged: (String? newEngine) {
                          ttsService.changedEngines(newEngine);
                            setState(() {
                              ttsService.engine = newEngine;
                            });
                        });
```

delee-dix replied:
작성해주신 콜백 함수를 적용하니 잘 작동하는 것을 확인했습니다!
아래에 언어 설정 하는 것 또한 같은 이슈가 있어 함께 수정 완료 하였습니다!

delee-dix committed:
[edit] tts_example

• main.dart → stt_tts_app.dart 이름 수정
• StatefulWidget → StatelessWidget
• tts_engine, tts_language
→ ui가 안 보이는 부분 수정
• 핸들러 상태 표시 위해 print() 추가 → console에서 상태 확인 가능
